docs/design/phases.md

```diff
@@ -0,0 +1,127 @@
+## Phases in kubeadm
+
+`kubeadm init` does a lot. In order to make kubeadm the common building block for Kubernetes, each logical phase kubeadm does during `init` should be invokable separately. 
+Here's an explanation what kubeadm does in these different phases:
+
+#### A note on constants / well-known values and paths
+
+We have to draw the line somewhere about what should be configurable, what shouldn't, and what should be hard-coded in the binary.
+
+We've decided to make the Kubernetes directory /etc/kubernetes a constant in the application, since it is clearly the given path in a majority of cases,
+and the most intuitive location. Having that path configurable would confuse readers of an on-top-of-kubeadm-implemented deployment solution.
+
+Further, having the admin/kubelet kubeconfig path as constants as well (`/etc/kubernetes/admin.conf` and `/etc/kubernetes/kubelet.conf`) would make things easier as well.
+Is it ok to have `/etc/kubernetes/kubelet.conf` as a "standard" between official deployments? It defines the path for the kubelet's KubeConfig file
+
+### Certificates
+
+First off, kubeadm generates certificate and private key pairs for different purposes.
+Certificates are stored by default in `/etc/kubernetes/pki`.
+
+There should be:
+ - a CA certificate (`ca.crt`) with its private key (`ca.key`)
+ - an API Server certificate (`apiserver.crt`) using `ca.crt` as the CA with its private key (`apiserver.key`). The certificate should:
+   - be a serving server certificate (`x509.ExtKeyUsageServerAuth`)
+   - contain altnames for
+     - the kubernetes' service's internal clusterIP and dns name (e.g. `10.96.0.1`, `kubernetes.default.svc.cluster.local`, `kubernetes.default.svc`, `kubernetes.default`, `kubernetes`)
+     - the hostname of the node
+     - the IPv4 address of the default route
+     - optional extra altnames that can be specified by the user
```

Comment:
This has been possible from the beginning by using `--api-external-dns-names`, but it will probably be exposed in an other shape in the future (via the API Group)